libs/api-provider-environment/src/transaction/useTransactionQueue.tsx

```diff
@@ -0,0 +1,201 @@
+import {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import {
+  TransactionItemStatus,
+  TransactionPayload,
+} from '@webb-tools/webb-ui-components';
+import {
+  Transaction,
+  TransactionState,
+  TransactionStatusValue,
+} from '@webb-tools/abstract-api-provider';
+import { TokenIcon } from '@webb-tools/icons';
+
+function transactionItemStatusFromTxStatus<Key extends TransactionState>(
+  txStatus: TransactionState
+): TransactionItemStatus {
+  switch (txStatus) {
+    case TransactionState.Done:
+    case TransactionState.Ideal:
+      return 'completed';
+    case TransactionState.Failed:
+      return 'warning';
+    default:
+      return 'in-progress';
+  }
+}
+function mapTxToPayload(
+  tx: Transaction<any>,
+  dismissTransaction: (id: string) => void
+): TransactionPayload {
+  const [txStatus, data] = tx.currentStatus;
+  const { amount, wallets, token, tokens } = tx.metaData;
+  return {
+    id: tx.id,
+    txStatus: {
+      status: transactionItemStatusFromTxStatus(txStatus),
+      message: getTxMessageFromStatus(txStatus, data),
+      txHash: tx.txHash,
+    },
+    amount: String(amount),
+    getExplorerURI(addOrTxHash: string, variant: 'tx' | 'address'): string {
+      return `https://explorer.moonbeam.network/${variant}/${addOrTxHash}`;
+    },
+    timestamp: tx.timestamp,
+    token,
+    tokens: tokens,
+    wallets: {
+      src: <TokenIcon size={'lg'} name={wallets.src || 'default'} />,
+      dist: <TokenIcon size={'lg'} name={wallets.dist || 'default'} />,
+    },
+    onDismiss(): void {
+      return dismissTransaction(tx.id);
+    },
+
+    method: tx.name as any,
+  };
+}
+function getTxMessageFromStatus<Key extends TransactionState>(
+  txStatus: Key,
+  transactionStatusValue: TransactionStatusValue<Key>
+): string {
+  switch (txStatus) {
+    case TransactionState.Cancelling:
+      return 'Canceling the Transaction';
```

Comment:
```suggestion
      return 'Canceling transaction';
```

Reply:
I'm curious why don't you update this message?